### ISSUE TITLE
AttributeError: module 'tensorflow.compat' has no attribute 'v1'

### ISSUE THREAD

**Issue by goeckeritz** [oushujun/EDTA#146]
Hi Shujun, 

I love your comprehensive pipeline!! Thank you for the easy installation tips too (I used the step-by-step).

I'm running the full pipeline in order to mask repeats before annotation using MAKER-P. I'd also generally like the annotation files for as many types of repeats as possible for those interested repeat dynamics in the future...

Anyway! The LTR and Helitron identification seemed to go perfectly fine, but TIR started and crashed. My error file shows the following: 

``Wed Jan  6 18:10:41 EST 2021    Identify TIR candidates from scratch.

Species: others
Traceback (most recent call last):
  File "/mnt/ufs18/home-087/goeckeri/EDTA/bin/TIR-Learner2.5/Module3_New/getDataset.py", line 17, in <module>
    tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
AttributeError: module 'tensorflow.compat' has no attribute 'v1'
cat: *-+-DTA.fa: No such file or directory
cat: *-+-DTC.fa: No such file or directory
cat: *-+-DTH.fa: No such file or directory
cat: *-+-DTM.fa: No such file or directory
cat: *-+-DTT.fa: No such file or directory
cat: *-+-NonTIR.fa: No such file or directory
cat: *-+-*-+-*.gff3: No such file or directory
rm: cannot remove ‘*-+-*-+-*.gff3’: No such file or directory
Traceback (most recent call last):
  File "/mnt/ufs18/home-087/goeckeri/EDTA/bin/TIR-Learner2.5/Module3_New/CombineAll.py", line 75, in <module>
    f_m3=removeDupinSingle("%s.gff3"%(genome_Name+spliter+"Module3"))
  File "/mnt/ufs18/home-087/goeckeri/EDTA/bin/TIR-Learner2.5/Module3_New/CombineAll.py", line 57, in removeDupinSingle
    f=pd.read_csv(file,header=None,sep="\t") #shujun
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/site-packages/pandas/io/parsers.py", line 702, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/site-packages/pandas/io/parsers.py", line 429, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/site-packages/pandas/io/parsers.py", line 895, in __init__
    self._make_engine(self.engine)
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/site-packages/pandas/io/parsers.py", line 1122, in _make_engine
 self._engine = CParserWrapper(self.f, **self.options)
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/site-packages/pandas/io/parsers.py", line 1853, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 545, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/mnt/ufs18/home-087/goeckeri/EDTA/bin/TIR-Learner2.5/Module3/GetAllSeq.py", line 32, in GetListFromFile
    f=open(file,"r+")
FileNotFoundError: [Errno 2] No such file or directory: 'TIR-Learner_FinalAnn_filter.gff3'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/mnt/ufs18/home-087/goeckeri/EDTA/bin/TIR-Learner2.5/Module3/GetAllSeq.py", line 63, in <module>
    pool.map(GetListFromFile,fileList) #shujun
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/multiprocessing/pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/opt/software/Python/3.6.4-foss-2018a/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
FileNotFoundError: [Errno 2] No such file or directory: 'TIR-Learner_FinalAnn_filter.gff3'
mv: cannot stat ‘TIR-Learner/*FinalAnn*.gff3’: No such file or directory
mv: cannot stat ‘TIR-Learner/*FinalAnn*.fa’: No such file or directory
Can't open ./TIR-Learner-Result/TIR-Learner_FinalAnn.fa: No such file or directory at /mnt/ufs18/home-087/goeckeri/EDTA/util/rename_tirlearner.pl line 19.
Warning: LOC list Pcerasus_haploidA.fasta.mod.TIR.ext30.list is empty.

Error: Error while loading sequenceCan't open ./TIR-Learner-Result/TIR-Learner_FinalAnn.gff3: No such file or directory.
Warning: The TIR result file has 0 bp!

Thu Jan  7 09:31:01 EST 2021    Start to find Helitron candidates.

Thu Jan  7 09:31:01 EST 2021    Identify Helitron candidates from scratch.

Thu Jan  7 17:35:48 EST 2021    Finish finding Helitron candidates.

Thu Jan  7 17:35:48 EST 2021    Execution of EDTA_raw.pl is finished!

ERROR: Raw TIR results not found in Pcerasus_haploidA.fasta.mod.EDTA.raw/Pcerasus_haploidA.fasta.mod.TIR.raw.fa
        If you believe the program is working properly, this may be caused by the lack of intact TIRs in your genome. Consider to use the --force 1 parameter to overwrite this check``

Any idea what might be the issue here? I fear it's a numpy version problem or something, but I should be using the recommended version in the step-by-step installation instructions. (I do pip3 list and numpy 1.16.4 shows up!)

Thanks so much in advance for your help!!

Kindly, 
Charity

**Comment by goeckeritz** [comment]
I also wanted to mention that the test case appeared to run just fine -- here's the log file for that:

``
Wed Jan  6 10:56:37 EST 2021    Dependency checking:
                                All passed!

        A custom library ../database/rice6.9.5.liban is provided via --curatedlib. Please make sure this is a manually curated library but not machine generated.

        A CDS file genome.cds.fa is provided via --cds. Please make sure this is the DNA sequence of coding regions only.

        A BED file is provided via --exclude. Regions specified by this file will be excluded from TE annotation and masking.

Wed Jan  6 10:56:38 EST 2021    Obtain raw TE libraries using various structure-based programs:
Wed Jan  6 10:57:43 EST 2021    Obtain raw TE libraries finished.
                                All intact TEs found by EDTA:
                                        genome.fa.mod.EDTA.intact.fa
                                        genome.fa.mod.EDTA.intact.gff3

Wed Jan  6 10:57:43 EST 2021    Perform EDTA advcance filtering for raw TE candidates and generate the stage 1 library:

Wed Jan  6 10:58:17 EST 2021    EDTA advcance filtering finished.

Wed Jan  6 10:58:17 EST 2021    Perform EDTA final steps to generate a non-redundant comprehensive TE library:

                                Use RepeatModeler to identify any remaining TEs that are missed by structure-based methods.

Wed Jan  6 10:59:33 EST 2021    Clean up TE-related sequences in the CDS file with TEsorter:

                                Remove CDS-related sequences in the EDTA library:

Wed Jan  6 10:59:49 EST 2021    Combine the high-quality TE library rice6.9.5.liban with the EDTA library:

Wed Jan  6 10:59:56 EST 2021    EDTA final stage finished! You may check out:
                                The final EDTA TE library: genome.fa.mod.EDTA.TElib.fa
                                Family names of intact TEs have been updated by rice6.9.5.liban: genome.fa.mod.EDTA.intact.gff3
                                Comparing to the provided library, EDTA found these novel TEs: genome.fa.mod.EDTA.TElib.novel.fa
                                The provided library has been incorporated into the final library: genome.fa.mod.EDTA.TElib.fa

Wed Jan  6 10:59:56 EST 2021    Perform post-EDTA analysis for whole-genome annotation:

Wed Jan  6 11:00:04 EST 2021    TE annotation using the EDTA library has finished! Check out:
                                Whole-genome TE annotation (total TE: 35.78%): genome.fa.mod.EDTA.TEanno.gff3
                                Whole-genome TE annotation summary: genome.fa.mod.EDTA.TEanno.sum
                                Low-threshold TE masking for MAKER gene annotation (masked: 16.32%): genome.fa.mod.MAKER.masked

Wed Jan  6 11:00:04 EST 2021    Evaluate the level of inconsistency for whole-genome TE annotation (slow step):

Wed Jan  6 11:00:22 EST 2021    Evaluation of TE annotation finished! Check out these files:

                                Overall: genome.fa.mod.EDTA.TE.fa.stat.all.sum
                                Nested: genome.fa.mod.EDTA.TE.fa.stat.nested.sum
                                Non-nested: genome.fa.mod.EDTA.TE.fa.stat.redun.sum
``

AND, if this helps, this is my pip3 list output: 


``
Package              Version
-------------------- -------------------
absl-py              0.11.0
astor                0.8.1
biopython            1.78
certifi              2020.12.5
dill                 0.3.3
drmaa                0.7.9
gast                 0.4.0
GenomeTools          1.6.1
glob2                0.4.1
google-pasta         0.2.0
grpcio               1.34.0
h5py                 2.10.0
importlib-metadata   3.3.0
Keras                2.2.4
Keras-Applications   1.0.8
Keras-Preprocessing  1.1.0
Mako                 1.1.3
Markdown             3.3.3
MarkupSafe           1.1.1
multiprocess         0.70.11.1
numpy                1.16.4
pandas               1.1.3
pip                  20.3.3
pp                   1.6.4.4
protobuf             3.14.0
pygpu                0.7.6
python-dateutil      2.8.1
pytz                 2020.5
PyYAML               5.3.1
regex                2016.6.24
scikit-learn         0.19.0
scipy                0.19.1
setuptools           49.6.0.post20201009
six                  1.15.0
tensorboard          1.14.0
tensorflow           1.14.0
tensorflow-estimator 1.14.0
termcolor            1.1.0
TEsorter             1.3
Theano               1.0.3
typing-extensions    3.7.4.3
Werkzeug             1.0.1
wheel                0.36.2
wrapt                1.12.1
zipp                 3.4.0
``

**Comment by oushujun** [comment]
Try #137 ?

**Comment by oushujun** [comment]
I see you already did. Let me see...

**Comment by oushujun** [comment]
Try to downgrade tf to 1.13? https://github.com/tensorflow/tensorflow/issues/24524

**Comment by goeckeritz** [comment]
Hey Shujun,

I'll downgrade and relaunch with overwrite = 0. I'll keep you posted!!

**Comment by goeckeritz** [comment]
Hi Shujun, 

I downgraded to tensorflow 1.13.1 and I still received the same error in the subject line... if you have any other ideas I'm all ears! I'll continue investigating!

-Charity

**Comment by oushujun** [comment]
Something is still wrong with tf. Have you tried the conda one-liner installation plus #137?

**Comment by oushujun** [comment]
Also try to do the installation on a compute node. There are some limitations set on the head node.

**Comment by goeckeritz** [comment]
Hi Shujun, 

You mean the quick installation one-liner installation, right? I'll give all of these a shot at once! I'll update you soon... Thank you!!

**Comment by mscharmann** [comment]
Hi, I think I have the same Problem; I though it could be this:
https://github.com/tensorflow/tensorflow/issues/44467 ?
But then I downgraded to h5py==2.10.0 and the problem remains..

**Comment by goeckeritz** [comment]
Hi mscharmann, 

Did you solve your issue? Are you trying to run EDTA on a cluster? I figured out what the issue was for me. Turns out my installation, either the conda one-liner or the conda step-by-step was working fine -- but yes, you do have to downgrade h5py if you catch it at a higher version. The issue was that I would activate my environment, but some modules would automatically be loaded in the shell I activated my environment in, and would unknowingly interfere. So I would highly encourage you to add 'module purge' or the command's equivalent to your job script or before loading your environment to remove any modules that could be messing up your environment (and looks like they tend to mess up tensorflow).

I didn't know environments tended to act this way... that was a fun week. Thanks Shujun for all of your help, by the way!!

Hope this helps!
-Charity
